generators/client/index.js

```diff
@@ -486,7 +486,7 @@ module.exports = JhipsterClientGenerator.extend({
             this.template(ANGULAR_DIR + 'components/util/_jhi-item-count.directive.js', ANGULAR_DIR + 'components/util/jhi-item-count.directive.js', this, {});
 
             // interceptor code
-            if (this.authenticationType === 'oauth2' || this.authenticationType === 'jwt') {
+            if (this.authenticationType == 'oauth2' || this.authenticationType == 'jwt' || this.applicationType == 'gateway') {
```

Comment:
why is `this.applicationType == 'gateway'` checked here? gateway will anyway have `jwt` and existing check should be sufficient rite? did i miss anything?